bin/problem.py

```diff
@@ -419,38 +421,33 @@ def has_constraints_checking(f):
                 warn(
                     f'No {validator_type} constraint validators found: No matches for \'constraints_file\'.'
                 )
-                return False
+                ok = False
 
             paths = constraint_validators
 
-        if validator_type == 'input_format':
-            validators = [
-                validate.InputValidator(
-                    problem,
-                    path,
-                    skip_double_build_warning=check_constraints,
-                    check_constraints=check_constraints,
-                )
-                for path in paths
-            ]
-        else:
-            validators = [
-                validate.OutputValidator(
-                    problem,
-                    path,
-                    skip_double_build_warning=check_constraints,
-                    check_constraints=check_constraints,
-                )
-                for path in paths
-            ]
+        validator_dispatcher = {
+                'input': validate.InputValidator,
+                'answer': validate.AnswerValidator,
+                'output': validate.OutputValidator
+                }
+        skip_double_build_warning = check_constraints or not paths_for_type['answer']
```

Comment:
I think the `check_constraints or` part of this actually isn't needed anymore. This was needed because I was passing some custom `-Dxyz` flags to the C++ compiler for constraints checking but those are now not used anymore.

Reply:
I didn’t want to touch this.

Reply:
Fair enough; I'll make an issue to clean this up later.